src/dmd/dscope.d

```diff
@@ -57,6 +57,7 @@ enum SCOPE
     ignoresymbolvisibility    = 0x0200,   /// ignore symbol visibility
                                           /// https://issues.dlang.org/show_bug.cgi?id=15907
     onlysafeaccess = 0x0400,  /// unsafe access is not allowed for @safe code
+    nosharedcheck  = 0x0800, /// allow shared acsses for the sub-expressions
```

Comment:
1. I don't care for flags that negate something when on. 
2. how is this related to `restrictiveshare`?
3. fix spelling